exchanges/kraken/kraken_websocket.go

```diff
@@ -583,156 +441,99 @@ func (k *Kraken) wsProcessOpenOrders(ownOrders interface{}) error {
 		}
 		return nil
 	}
-	return errors.New(k.Name + " - Invalid own trades data")
+	return errors.New("invalid own trades data")
 }
 
-// addNewSubscriptionChannelData stores channel ids, pairs and subscription types to an array
-// allowing correlation between subscriptions and returned data
-func (k *Kraken) addNewSubscriptionChannelData(response *wsSubscription) {
-	// We change the / to - to maintain compatibility with REST/config
-	var pair, fPair currency.Pair
-	var err error
-	if response.Pair != "" {
-		pair, err = currency.NewPairFromString(response.Pair)
-		if err != nil {
-			log.Errorf(log.ExchangeSys, "%s exchange error: %s", k.Name, err)
-			return
+// wsProcessTickers converts ticker data and sends it to the datahandler
+func (k *Kraken) wsProcessTickers(response []any, pair currency.Pair) error {
+	t, ok := response[1].(map[string]any)
+	if !ok {
+		return errors.New("received invalid ticker data")
+	}
+	data := make([]float64, 9)
```

Comment:
Seems like this only needs to be 7?

Reply:
Well if that's true then L480 is totally Fckd

Reply:
Why the hell didn't I use a map?
Significantly more sustainable code.

Fixed a279e5f6